src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/FilePathNormalizer.cs

```diff
@@ -19,9 +19,13 @@ internal static class FilePathNormalizer
 
     private class FilePathNormalizingComparer : IEqualityComparer<string>
     {
+        private Func<char, char> _charConverter = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+            ? c => c
+            : char.ToLowerInvariant;
```

Comment:
@davidwengier: Is invariant the right approach? After all, file paths are human-readable strings with a potential for important culture details being included in them. Asking because I'm unsure. 🤷

Reply:
I picked invariant because our existing conversions are `Ordinal`, which is culture-less, and given there is no `ToLowerOrdinal()`, avoiding the expense of culture seemed workwhile.